Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift

```diff
@@ -1,4 +1,313 @@
+    private var userId: String
+    private var jobId: String
+    private var document: Document
+    private var currentBuffer: CVPixelBuffer?
+    private (set) var frontImage: UIImage?
+    private (set) var backImage: UIImage?
+    private var subscribers = Set<AnyCancellable>()
+    private var cameraFeedSubscriber: AnyCancellable?
+    private var captureSubscriber: AnyCancellable?
+    private var captureBothSides: Bool
+    private let rectangleFunnel = RectangleFeaturesFunnel()
+    private (set) lazy var cameraManager: CameraManageable = CameraManager(orientation: .landscape)
+    private (set) var side = Side.front
+    private (set) var showAttribution: Bool
+    private var selfie: Data?
+    private var recieveBufferQueue = DispatchQueue(label: "com.smileid.receivebuffer")
+    @State var galleryImageFront = UIImage() {
+        didSet {
+            frontImage = galleryImageFront
+        }
+    }
+    @Published var processingState: ProcessingState? {
+        didSet {
+            switch processingState {
+            case .none:
+                resumeCameraSession()
+            case .endFlow:
+                pauseCameraSession()
+            case .some:
+                pauseCameraSession()
+            }
+        }
+    }
+
+    init(userId: String, jobId: String, document: Document, captureBothSides: Bool, showAttribution: Bool) {
+        self.userId = userId
+        self.jobId = jobId
+        self.document = document
+        self.captureBothSides = captureBothSides
+        self.showAttribution = showAttribution
+        subscribeToCameraFeed()
+        subscribeToImageCapture()
+    }
+
+    func subscribeToCameraFeed() {
+       cameraFeedSubscriber = cameraManager.sampleBufferPublisher
+            .receive(on: recieveBufferQueue)
+            .compactMap({$0})
+            .sink( receiveValue: { [self] buffer in
+                self.currentBuffer = buffer
+                let imageSize = CGSize(width: CVPixelBufferGetWidth(buffer),
+                                       height: CVPixelBufferGetHeight(buffer))
+                RectangleDetector.rectangle(forPixelBuffer: buffer,
+                                            aspectRatio: document.aspectRatio) { rect in
+                    self.processRectangle(rectangle: rect, imageSize: imageSize)
+                }
+            })
+    }
+
+    func subscribeToImageCapture() {
+        captureSubscriber = cameraManager.capturedImagePublisher
+            .receive(on: DispatchQueue.global())
+            .compactMap({$0})
+            .sink( receiveValue: { image in
+                DispatchQueue.main.async {
+                    self.processingState = .confirmation(image)
+                }
+            })
+    }
+
+    func captureImage() {
+        cameraManager.capturePhoto()
+    }
+
+    func resetState() {
+        DispatchQueue.main.async {
+            if self.processingState != nil {
+                self.processingState = nil
+            }
```

Comment:
This check seems redundant
```suggestion
            self.processingState = nil
```